subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -223,6 +225,10 @@ int bt_csip_set_member_generate_rsi(const struct bt_csip_set_member_svc_inst *sv
 		return res;
 	}
 
+	/* Make sure Little Endian before copying to output */
+	prand = sys_cpu_to_le32(prand);
+	hash = sys_cpu_to_le32(hash);
+
 	(void)memcpy(rsi, &hash, BT_CSIP_SIH_HASH_SIZE);
 	(void)memcpy(rsi + BT_CSIP_SIH_HASH_SIZE, &prand, BT_CSIP_SIH_PRAND_SIZE);
```

Comment:
Instead of making all the steps use little endian, couldn't we just keep them as host endianess until we copy it to the `rsi`? 

i.e. remove the calls to `sys_cpu_to_le32` and instead ensure that values are copied as little endian into `rsi`? 

I think that makes it easier to maintain. 

We should probably also documentat that the `rsi` array that we output is little endianess (and thus ready to be advertised), as we don't actually say that in the documentation anywhere.

Reply:
> Instead of making all the steps use little endian, couldn't we just keep them as host endianess until we copy it to the rsi?

There is actually just one step where we do it before hand, in SIH before we set the highest bits to b'01. The alternative is to instead use an if-else there depending on host-endianness. I'm not sure if I'm more fond of that. What do you think?
Or any other ideas?

> We should probably also documentat that the rsi array that we output is little endianess (and thus ready to be advertised), as we don't actually say that in the documentation anywhere.

That's a good point. Should we add this anywhere else but in the doxygen descripton in the header?